### ISSUE TITLE
Repeat for with primitive types - invalid contextual properties

### ISSUE THREAD

**Issue by aegenet** [aurelia/aurelia#2037]
# 🐛 Bug Report

I've just caught a bug with the repeat.for template controller: most contextual properties _($index, $first...)_ are invalid with primitive arrays _(number[], boolean[], string[]...)_.

```html
<ul>
  <!-- <li repeat.for="val of ['a','a','b','c','d','d','d','d','h']"> -->
  <!-- <li repeat.for="val of [true,true,false,false,false,true,false,true,true]"> -->
  <!-- <li repeat.for="val of [undefined,true,'a',false,false,'a',false,true,undefined]"> -->
  <li repeat.for="val of [1,1,2,3,4,4,4,4,5]">
    index: ${$index}
    first: ${$first}
    even: ${$even}
    value: ${val}
  </li>
</ul>
```

Repro link: https://gist.dumber.app/?gist=6a430716e4df710a5de3816124fa8018

## 🤔 Expected Behavior

```bash
index: 0; first: true; even: true; value: 1;
index: 1; first: false; even: false; value: 1;
index: 2; first: false; even: true; value: 2;
index: 3; first: false; even: false; value: 3;
index: 4; first: false; even: true; value: 4;
index: 5; first: false; even: false; value: 4;
index: 6; first: false; even: true; value: 4;
index: 7; first: false; even: false; value: 4;
index: 8; first: false; even: true; value: 5;
```

## 😯 Current Behavior

```bash
index: 1; first: false; even: false; value: 1;
index: 1; first: false; even: false; value: 1;
index: 2; first: false; even: true; value: 2;
index: 3; first: false; even: false; value: 3;
index: 7; first: false; even: false; value: 4;
index: 7; first: false; even: false; value: 4;
index: 7; first: false; even: false; value: 4;
index: 7; first: false; even: false; value: 4;
index: 8; first: false; even: true; value: 5;
```

## 💁 Possible Solution

_Workaround: object array_

## 💻 Code Sample

Repro link: https://gist.dumber.app/?gist=6a430716e4df710a5de3816124fa8018

## 🌍 Your Environment

| Software         | Version(s) |
| ---------------- | ---------- |
| Aurelia          |  2.0.0-beta.1 to 2.0.0-beta.21
| Language         | 
| Browser          |
| Bundler          |
| Operating System |
| NPM/Node/Yarn    |

**Comment by ekzobrain** [comment]
Hm, I remember I reported this bug already and it seemed to be fixed by @fkleuver

**Comment by bigopon** [comment]
@ekzobrain it's here https://github.com/aurelia/aurelia/pull/1737
In that PR only the rendering part was fixed, we forgot the contextual properties part. Thanks @aegenet

**Comment by bigopon** [comment]
@ekzobrain you are actually right, it's supposed to be fixed in a previous PR from @fkleuver .
When I tried to create a failing test with the same template given in this issue, I couldn't reproduce the broken behaior, I'll have to investigate more.

**Comment by bigopon** [comment]
resolved by #2050, thanks @aegenet
